packages/flutter/lib/src/widgets/container.dart

```diff
@@ -105,6 +109,54 @@ class DecoratedBox extends SingleChildRenderObjectWidget {
   }
 }
 
+class _DynamicBox extends StatelessWidget {
+  /// This class determines which widget to build based on config's type.
+  const _DynamicBox(this.config, {required this.child});
+
+  final dynamic config;
+  final Widget child;
+
+  Never error(String description) => throw Exception('$description: $config');
+
+  @override
+  Widget build(BuildContext context) => switch (config) {
```

Comment:
This build method as written is hard to read and reason about. This doesn't feel like maintainable code.